domain/src/SharePooling/ValueObjects/SharePoolingTokenAcquisitions.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+namespace Domain\SharePooling\ValueObjects;
+
+use ArrayIterator;
+use Domain\SharePooling\ValueObjects\Exceptions\SharePoolingTransactionException;
+use Domain\ValueObjects\FiatAmount;
+use Domain\ValueObjects\Quantity;
+use IteratorAggregate;
+use Traversable;
+
+/** @implements IteratorAggregate<int, SharePoolingTokenAcquisition> */
+final class SharePoolingTokenAcquisitions implements IteratorAggregate
+{
+    /** @param array<int, SharePoolingTokenAcquisition> $acquisitions */
+    private function __construct(private array $acquisitions = [])
+    {
+    }
+
+    public static function make(SharePoolingTokenAcquisition ...$acquisitions): self
+    {
+        return new self(array_values($acquisitions));
+    }
+
+    /** @return Traversable<int, SharePoolingTokenAcquisition> */
+    public function getIterator(): Traversable
+    {
+        return new ArrayIterator($this->acquisitions);
+    }
+
+    public function isEmpty(): bool
+    {
+        return empty($this->acquisitions);
+    }
+
+    public function count(): int
+    {
+        return count($this->acquisitions);
+    }
+
+    public function add(SharePoolingTokenAcquisition ...$acquisitions): self
+    {
+        foreach ($acquisitions as $acquisition) {
+            try {
+                $acquisition->setPosition($this->count());
+            } catch (SharePoolingTransactionException) {
+            }
+
+            $position = $acquisition->getPosition();
+
+            assert(! is_null($position));
+
+            $this->acquisitions[$position] = $acquisition;
+        }
```

Comment:
ℹ️ This ensures a transaction is always inserted at the same position, so the chronological order is maintained without having to keep track of the transactions' time